packages/transaction-controller/src/utils/utils.ts

```diff
@@ -158,8 +158,8 @@ export function normalizeTxError(
     name: error.name,
     message: error.message,
     stack: error.stack,
-    code: error?.code,
-    rpc: error?.value,
+    code: error.code,
+    rpc: error.value ? JSON.parse(JSON.stringify(error.value)) : undefined,
```

Comment:
does it matter if falsey like `0` or `''` or `null` would get folded into `undefined`?

Reply:
Good call. I've updated this so that we use `error.value` as-is if it's JSON-encodable, otherwise we use `undefined`: [f20e8fd](https://github.com/MetaMask/core/pull/3827/commits/f20e8fd4b8c92afaf651e9b7e2b30d69e42a22a7)